ext/zip/php_zip.c

```diff
@@ -1707,8 +1698,13 @@ static void php_zip_add_from_pattern(INTERNAL_FUNCTION_PARAMETERS, int type) /*
 					file_stripped = ZSTR_VAL(basename);
 					file_stripped_len = ZSTR_LEN(basename);
 				} else if (remove_path && strstr(Z_STRVAL_P(zval_file), remove_path) != NULL) {
-					file_stripped = Z_STRVAL_P(zval_file) + remove_path_len;
-					file_stripped_len = Z_STRLEN_P(zval_file) - remove_path_len;
+					if (Z_STRVAL_P(zval_file)[remove_path_len] == '/' || Z_STRVAL_P(zval_file)[remove_path_len] == '\\') {
```

Comment:
I think this could use [`IS_SLASH()`](https://github.com/php/php-src/blob/e786e9c83c56613c861f431451526a6a27d5dc7c/Zend/zend_virtual_cwd.h#L75).

Reply:
indeed, done